dash.js

```diff
@@ -699,7 +699,20 @@ const MyDash = new Lang.Class({
     _redisplay: function() {
         let favorites = AppFavorites.getAppFavorites().getFavoriteMap();
 
-        let running = this._appSystem.get_running();
+        let running;
+        if (this._dtdSettings.get_boolean('isolate-workspaces')) {
+            running = this._appSystem.get_running().filter(function(_app) {
```

Comment:
Absolutely. Thanks for the suggestion. I had to redeclare 'settings' to use them. See patch [85cfc760aba7f3a5a2c1545d45bed8f1c0d94577].

I was thinking that maybe it would be nicer to write `getInterestingWindows()` so that it takes the boolean instead of the settings.
That way if at some point you need to get ALL windows no matter the settings, you can simply pass `getInterestingWIndows(app, true)`.

In this case, we would write:
`return AppIcons.getInterestingWindows(_app, true).length != 0`

What do you think? Both ways it's fine, just a detail I guess.